hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/strategy/BoundedIOCompactionStrategy.java

```diff
@@ -46,7 +47,7 @@ public List<HoodieCompactionOperation> orderAndFilter(HoodieWriteConfig writeCon
       targetIORemaining -= opIo;
       finalOperations.add(op);
       if (targetIORemaining <= 0) {
-        return finalOperations;
+        missingPartitions.add(op.getPartitionPath());
```

Comment:
Usually we do not modify the passed in collections.

Reply:
changed.

Reply:
Can we add some doc around the return values of `#orderAndFilter`

Reply:
added